### PR TITLE
fix: update dev environment, and add CI

### PR DESCRIPTION
**Pull request by TarikGul**
[Do Not Merge]: There are a few things I would like to iron out before merging. 

## What is Added:

1. `@substrate/dev` has been added as a dev dep and all its eslint, prettier, and tsconfigs specs are available to txintegration. 

2. Added a minimum version for node which is 14, this is a specification based off of polkadot-js. 
    a. Added the `.nvmrc` file
    b. Added "engines" field in package.json

3. Added a `tsconfig.build.json`

4. Added `.eslintrc.js` and `.eslintignore`
    - Depending on what you want we can either 
    
    a. inherit the same specs we use in txwrapper-core and sidecar for linting. 
    b. choose your own specs. Regardless we should lint the repo after this PR and let the CI test fail in order to not cloud up this PR

5. Added CI checks for `lint` and `build`.

6. throw an error when metadata, or specName is not provided.

### CODE REVIEW

**Comment by joepetrowski** [comment]
Nice, thanks! I'd use the same specs from txwrapper/sidecar.

File: src/sign.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ function getSigningInfo(): SigningInfo {
   if (signingInfo.specName === 'polkadot'){ chainName = 'Polkadot'; }
   else if (signingInfo.specName === 'kusama'){ chainName = 'Kusama'; }
   else if (signingInfo.specName === 'westend'){ chainName = 'Westend'; }
-  else { console.warn(`Error, registry for ${signingInfo.specName} not supported.`); }
```
Comment (TarikGul) [inline src/sign.ts:30]: Switched from a console.warn to a throw. When the specName doesn't exist the program should stop execution. (Part of why the typescript build was failing).

File: src/sign.ts [hunk 2]
```diff
@@ -63,13 +63,11 @@ async function main(): Promise<void> {
     }
     case 'westend': {
       SS58_FORMAT = 42;
-      console.warn(`TODO: Need metadata for Westend. Likely to get BadProof error.`)
-      break;
+      throw Error(`TODO: Need metadata for Westend. Likely to get BadProof error.`);
```
Comment (TarikGul) [inline src/sign.ts:66]: Same as the specName, when the metadata doesnt exist it should throw not warn.

**Comment by TarikGul** [comment]
@joepetrowski Should be ready to merge whenever you are ready. Following this PR, I'll have a linting PR that cleans up all the `any` types.

**Review by joepetrowski — APPROVED** [review]

**Pull request merged**